scripts/global.js

```diff
@@ -613,45 +628,178 @@ export function createActivityListHTML(arrTXs, fRewards = false) {
+                if (
+                    !mapOurAddresses.has(strAddr) &&
+                    !(await isYourAddress(strAddr))[0]
+                ) {
+                    // External address, this is not a self-only Tx
+                    fSendToSelf = false;
+                } else {
+                    // Internal address, remember this for later use
+                    mapOurAddresses.set(strAddr);
+                }
+            }
+        }
+
+        // Generate an icon, colour and description for the Tx
+        if (!fRewards) {
+            switch (cTx.type) {
+                case HistoricalTxType.STAKE:
+                    icon = 'fa-gift';
+                    break;
+                case HistoricalTxType.SENT:
+                    icon = 'fa-minus';
+                    colour = '#f93c3c';
+                    // Figure out WHO this was sent to, and focus on them contextually
+                    if (fSendToSelf) {
+                        txContent = 'Sent to self';
+                    } else {
+                        // Otherwise, anything to us is likely change, so filter it away
+                        const arrExternalAddresses = cTx.receivers.filter(
+                            (addr) => !mapOurAddresses.has(addr)
+                        );
+                        txContent =
+                            'Sent to ' +
+                            (cTx.shieldedOutputs
+                                ? 'Shielded address'
+                                : [
+                                      ...new Set(
+                                          arrExternalAddresses.map((addr) =>
+                                              addr.length >= 32
+                                                  ? addr.substring(0, 6)
+                                                  : addr
+                                          )
+                                      ),
+                                  ].join(', ') + '...');
+                    }
+                    break;
+                case HistoricalTxType.RECEIVED: {
+                    icon = 'fa-plus';
+                    colour = '#5cff5c';
+                    // Figure out WHO this was sent from, and focus on them contextually
+                    // Filter away any of our own addresses
+                    const arrExternalAddresses = cTx.senders.filter(
+                        (addr) => !mapOurAddresses.has(addr)
+                    );
+                    if (cTx.shieldedOutputs) {
+                        txContent = 'Received from Shielded address';
+                    } else {
+                        txContent =
+                            'Received from ' +
+                            [
+                                ...new Set(
+                                    arrExternalAddresses.map((addr) =>
+                                        addr.length >= 32
```

Comment:
Some txs don't have an address (https://explorer.rockdev.org/tx/3ac32374b11eb30481ea17adca5f27909e41185f4c245d929ace7c1dcd06c4c7)
```suggestion
                                        addr?.length >= 32
```